src/program.rs

```diff
@@ -231,8 +231,9 @@ impl Program {
         // Only if successful, remove image from tracked images
         self.images.remove(self.index);
 
-        // Adjust our view
-        self.decrement(1)
+        // Moving the image automatically advanced to next image
+        // Adjust our view to reflect this
+        self.render()
```

Comment:
Before this, you'll need to check if the index is now past the length of the vector, otherwise this will crash on moving the last image.

Reply:
I added some debug info locally.
```rust
        // Only if successful, remove image from tracked images
        self.images.remove(self.index);

        // Moving the image automatically advanced to next image
        // Adjust our view to reflect this
        dbg!(self.images.len());
        self.render()
    }
```

I had two images found.
// After first removal
[src/program.rs:236] self.images.len() = 1
// After second removal
[src/program.rs:236] self.images.len() = 0
// Black screen now shown
Failed to move file: no images to move
Failed to move file: no images to move

Reply:
This is correct behaviour, right?

Reply:
Until we handle the error differently. I'd think so.

If moving an image fails (because there are none, or another eason), then maybe notify the user somehow besides printing to console?

Reply:
Yes, once I have the info bar implemented, we will be able to print there.

Reply:
https://github.com/gurgalex/riv/blob/8db5a08dc55f4128268b34d70cc10da85272ffde/src/program.rs#L205

I guess we could ignore move and delete keypresses when no images are left.

Reply:
This pull request still does crash the program if trying to delete or move the last image.

Reply:
Umm, do you have a backtrace?

Which platform are you on?

I wonder why it's not crashing on Linux for me.

Reply:
Deleting or moving the only image won't crash the program, but if there are several images and you delete or move the last one, it will crash.

Reply:
```
thread 'main' panicked at 'index out of bounds: the len is 398 but the index is 398', /rustc/2aa4c46cfdd726e97360c2734835aa3515e8c858/src/libcore/slice/mod.rs:2455:10
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::panicking::panic_bounds_check
   9: riv::program::Program::render
  10: riv::program::Program::run
  11: riv::main
  12: std::rt::lang_start::{{closure}}
  13: std::panicking::try::do_call
  14: __rust_maybe_catch_panic
  15: std::rt::lang_start_internal
  16: main
```

Reply:
Oh!

I never moved to the last image.